app/javascript/src/component_expander.js

```diff
@@ -61,44 +61,71 @@ const {
  * @param $node (jQuery) Html Element to be enhanced.
  * @param condif (Object) Key/value pairs for config (see above)
 **/
+
 class Expander {
+  #browserControlled;
   #config;
   #state;
   #id;
 
+  static STATE_OPEN = "open";
+  static NATIVE_CONTROL_ATTRIBUTE = "open";
+
   constructor($content, config) {
     const conf = mergeObjects({
       activator_source: "activate", // Developer mistake occured if see this text.
-      wrap_content: true,
+      wrap_content: true && !$content.is("details"), // Default is true unless detect <details> in use.
       auto_open: false,
       duration: 0,
     }, config);
 
     this.#config = conf;
-    this.#state = 'open';
+    this.#state = ""; // This value always get overwritten by openState
     this.#id = uniqueString("Expander_");
+    this.#browserControlled = false; // Will be changed in createNode if $content is <details> element
     this.$node = this.#createNode($content);
     this.$activator = this.#createActivator();
 
     conf.auto_open ? this.open() : this.close();
   }
 
-  open() {
-    this.$node.slideDown( this.#config.duration, () => {
+  set openState(open) {
+    if(open) {
       this.$activator.attr("aria-expanded", "true");
-      this.#state = 'open';
-    });
+      this.#state = Expander.STATE_OPEN;
+    }
+    else {
+      this.$activator.attr("aria-expanded", "false");
+      this.#state = 'closed';
```

Comment:
I added the STATE_OPEN simply because it was a little confusing seeing "open" for that value and "open" for the attribute value in close proximity. Felt too easy to confuse the two and introduce a simple error. That's why I also put in the NATIVE_CONTROL_ATTRIBUTE. We do use it twice, so there is also some gain in using a constant in the traditional sense as well, but that was not a driving factor.

I nearly put in STATE_CLOSED for the consistency like you suggest, but didn't bother because the value of what would be STATE_CLOSED is somewhat irrelevant.

The internal state tracking value is never exposed. We're interested (inside isOpen() method only) to find if state is equal to STATE_OPEN but the "closed" value could just as well be an empty string, or "fred", or "anything else". In fact, if I'd used an empty string it might be more correct and you may not have even queried it.

Reply:
Understand the reasoning, but I actually still think it would be slightly better with it even if it is slightly redundant. Mostly because as it is, reading the code makes me have to think *why* is it not consistent and reason it out, whereas if both were there it would be obvious and consistent (if a little OTT).